agent/config/config_test.go

```diff
@@ -187,10 +163,23 @@ func TestConfigBoolean(t *testing.T) {
 func TestBadLoggingDriverSerialization(t *testing.T) {
 	os.Setenv("ECS_AVAILABLE_LOGGING_DRIVERS", "[\"malformed]")
 
-	conf := environmentConfig()
+	conf, err := environmentConfig()
+	assert.Nil(t, err)
 	if len(conf.AvailableLoggingDrivers) != 0 {
 		t.Error("Wrong value for AvailableLoggingDrivers", conf.AvailableLoggingDrivers)
 	}
+	// Unset env var so this invalid value doesn't corrupt later tests.
+	os.Unsetenv("ECS_AVAILABLE_LOGGING_DRIVERS")
```

Comment:
(multiple occurrences) The preferred way to do this would be with a `defer` right under where it's set

```go
os.Setenv("ECS_AVAILABLE_LOGGING_DRIVERS", `["malformed]`)
defer os.Unsetenv("ECS_AVAILABLE_LOGGING_DRIVERS")
```